postgres/tests/test_statements.py

```diff
@@ -1468,6 +1476,57 @@ def _sample_key(e):
     assert len(matching) > 0, "should have collected exactly at least one matching event"
 
 
+@pytest.mark.parametrize("pg_stat_activity_view", ["pg_stat_activity"])
+@pytest.mark.parametrize(
+    "query_samples_enabled,expected_samples_len_check", [(True, lambda a: a >= 1), (False, lambda a: a == 0)]
+)
+@pytest.mark.parametrize(
+    "query_activity_enabled,expected_activity_len_check", [(True, lambda a: a >= 1), (False, lambda a: a == 0)]
+)
+@pytest.mark.parametrize(
+    "user,password,dbname,query,arg",
+    [("bob", "bob", "datadog_test", "BEGIN TRANSACTION; SELECT city FROM persons WHERE city = %s;", "hello")],
+)
+def test_disabled_activity_or_explain_plans(
+    aggregator,
+    integration_check,
+    dbm_instance,
+    query_activity_enabled,
+    query_samples_enabled,
+    expected_samples_len_check,
+    expected_activity_len_check,
+    pg_stat_activity_view,
+    user,
+    password,
+    dbname,
+    query,
+    arg,
+):
+    """
+    Test four combinations for the following:
+        if activity sampling is enabled, ensure there are activity logs; else ensure there are none.
+        if explain plans are enabled (query_samples), ensure there are explain plan logs; else ensure there are none.
+    """
+    dbm_instance['pg_stat_activity_view'] = pg_stat_activity_view
+    dbm_instance['query_activity']['enabled'] = query_activity_enabled
+    dbm_instance['query_samples']['enabled'] = query_samples_enabled
+    check = integration_check(dbm_instance)
+    check._connect()
+
+    conn = psycopg2.connect(host=HOST, dbname=dbname, user=user, password=password)
+
+    try:
+        conn.cursor().execute(query, (arg,))
+        check.check(dbm_instance)
+        dbm_samples = aggregator.get_event_platform_events("dbm-samples")
+        dbm_activity = aggregator.get_event_platform_events("dbm-activity")
+
+        assert expected_activity_len_check(len(dbm_activity)) is True
+        assert expected_samples_len_check(len(dbm_samples)) is True
```

Comment:
I would not say this is necessarily a blocking comment but this test logic is too convoluted for my taste. There is potential for bugs and maintainability drift affecting the test over time. Passing in a lambda function that changes behavior depending on the configuration parameter is hard to follow and debug. I think simple `if` statements would be better suited here.